File: scheduler-rest-api/src/main/java/ua/ivan909020/scheduler/rest/service/task/TaskService.java
package ua.ivan909020.scheduler.rest.service.task;

import java.util.List;

import ua.ivan909020.scheduler.core.model.domain.page.KeysetPage;
import ua.ivan909020.scheduler.core.model.domain.page.PagedList;
import ua.ivan909020.scheduler.core.model.entity.TaskStatus;
import ua.ivan909020.scheduler.rest.model.dto.task.TaskDto;

public interface TaskService {

    PagedList<TaskDto> findAll(List<TaskStatus> statuses, KeysetPage page);

}

File: scheduler-ui/src/app/task/task.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { PagedList, Task, TaskStatus } from './task';
import { catchError, Observable, retry, throwError } from 'rxjs';
import { AppService } from '../app.service';

@Injectable()
export class TaskService {

  constructor(private http: HttpClient) {
  }

  findAll(statuses: TaskStatus[], pageSize: number, pageCursor?: string): Observable<PagedList<Task>> {
    const url: string = AppService.basePath + "/api/v1/tasks";

    let params = new HttpParams();
    params = params.append("statuses", statuses.join(','));
    params = params.append("pageSize", pageSize);
    if (pageCursor != null) {
      params = params.append("pageCursor", pageCursor);
    }

    return this.http.get<PagedList<Task>>(url, { params })
      .pipe(retry(1), catchError(error => throwError(() => error)));
  }

}
